cmd/validate/image.go

```diff
@@ -409,12 +412,17 @@ func validateImageCmd(validate imageValidationFunc) *cobra.Command {
 					e := fmt.Errorf("error validating image %s of component %s: %w", r.component.ContainerImage, r.component.Name, r.err)
 					allErrors = errors.Join(allErrors, e)
 				} else {
+					for key, val := range r.data {
```

Comment:
I don't think this is quite right. This is merging all the data from different source groups.

Consider something like this:

```go
// Renamed from manyData
var evaluatorData []evaluator.Data
dataCollected := false

for i := 0; i < numComponents; i++ {
  r := <-results
  ...
  if !dataCollected {
    evaluatorData = r.data
    dataCollected = true
  }
}
```

Reply:
or rather:

```go
// Renamed from manyData
var evaluatorData []evaluator.Data

for i := 0; i < numComponents; i++ {
  r := <-results
  ...
  if evaluatorData == nil {
    evaluatorData = r.data
  }
}
```

Anyways, I think you get the idea.

Reply:
Don't we want that since the data could differ per source group?

Reply:
`r.data` is a list of data.

Reply:
I changed it to `map[string]evaluator.Data`. My thought was each evaluator maps to a source group, so to just collect data per evaluator.
https://github.com/enterprise-contract/ec-cli/pull/2181/files#diff-7f11215ba891ff8199fa09a412849a5414e416050d163c813f6499009bbe0b96R304
https://github.com/enterprise-contract/ec-cli/pull/2181/files#diff-ed62eb280d370e54af87c7fdb7026890b2efeefd1da359a2b28ecb76904cd86eR136

Reply:
This makes me think this whole PR could be replaced with 
```diff
--- a/cmd/validate/image.go
+++ b/cmd/validate/image.go
@@ -410,7 +410,9 @@ func validateImageCmd(validate imageValidationFunc) *cobra.Command {
                                        allErrors = errors.Join(allErrors, e)
                                } else {
                                        components = append(components, r.component)
-                                       manyData = append(manyData, r.data)
+                                       if len(manyData) == 0 {
+                                               manyData = append(manyData, r.data)
+                                       }
                                        manyPolicyInput = append(manyPolicyInput, r.policyInput)
                                }
                        }


```